jaeger-core/src/main/java/io/jaegertracing/JaegerSpan.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016, Uber Technologies, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.jaegertracing;
+
+import io.opentracing.Span;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a Span as seen from Jaeger's perspective. Builds on OpenTracing's {@link Span}, adding properties that
+ * are not part of the standard.
+ *
+ * Should be used wisely by instrumented applications: always prefer OpenTracing's approach when available.
+ */
+public interface JaegerSpan extends Span {
+  Tracer getTracer();
```

Comment:
I would prefer not expose tracer here.

Reply:
I extracted `JaegerSpan` into an interface based on what other code is consuming. This PR is not about fixing those parts, but rather, in reorganizing the existing classes.

Reply:
> I extracted JaegerSpan into an interface based on what other code is consuming. This PR is not about fixing those parts, but rather, in reorganizing the existing classes.

There is a difference to what is needed in this repo(mostly tests) and what is exposed in general.

I don't see a value exposing a tracer in a span as the tracer is needed to start a span.

Reply:
There's zero difference between what this PR exposes and what the current code exposes, from a consumer's perspective. I'm not exposing anything *more* than before: code that had `io.jaegertracing.Span` before already had access to `getTracer()`.

This PR is really just about reorganizing the structure of the existing classes and is already too big to my taste. I do plan on sending more PRs after this one, moving things we don't intend to expose, such as this `getTracer()` method.

Work on defining the public/internal API will be mostly done as part of #396.